binderhub/builder.py

```diff
@@ -313,16 +313,18 @@ def escape(s):
             push_secret = None
 
         BuildClass = FakeBuild if self.settings.get('fake_build') else Build
-        binder_url = '{proto}://{host}{base_url}v2/{provider}/{spec}'.format(
-            proto=self.request.protocol,
-            host=self.request.host,
-            base_url=self.settings['base_url'],
-            provider=provider_prefix,
-            spec=spec,
-        )
+
+        binder_url = f"{self.request.protocol}://{self.request.host}{self.settings['base_url']}v2/" \
```

Comment:
nit: I find `{proto}://{host}{base_url}v2/{provider}/{spec}` easier to read than the longer version as f-string. If you don't have a strong preference I'd vote for sticking with the `.format()` style that lets us have shorter "templates".